src/Castle.Windsor.Extensions.MsDependencyInjection/NetCoreRootScope.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2004-2020 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Windsor.Extensions.MsDependencyInjection
+{
+    internal class NetCoreRootScope : NetCoreScope
+    {
+        internal static NetCoreRootScope RootScope {get; private set;}
```

Comment:
This static field means that there can only be one root scope for the process, is this right? And how does this get cleaned up?

Reply:
Yes - only once. And gets destroyed with last Service provider getting disposed. Its MS design decision.

Reply:
Thanks